lib/link.js

```diff
@@ -85,7 +85,7 @@ class Link {
     var type = value.toLowerCase()
 
     for( var i = 0; i < this.refs.length; i++ ) {
-      if( this.refs[ i ].rel.toLowerCase() === type ) {
+      if( this.refs[ i ].rel && this.refs[ i ].rel.toLowerCase() === type ) {
```

Comment:
Should this null- or typeof-check instead of relying on implicit falsy-ness? Thinking an empty `rel` could trip this up, e.g. `<example.com>; rel=`. Not sure if that's a real use-case, and don't remember if the specs would allow it though.

```suggestion
      if( typeof this.refs[ i ].rel == 'string' && this.refs[ i ].rel.toLowerCase() === type ) {
```

Reply:
Would also be a good opportunity to add test cases for `rel()`, `get()`, and `has()` with empty `rel`s in `test/api`, but I can also do that after merging this, so more of a note-to-self I guess :)

Reply:
I thought of typeof as well, and we can go with that. Though, isn't the lowerCase of empty string the same empty string? `rel` with empty value is not conforming as per RFC8288 but not sure if it matters here either.

Reply:
Perhaps change '===' instead of '=='. I've used the same approach for the others.

Reply:
> Though, isn't the lowerCase of empty string the same empty string? 

Yes, but it wouldn't get to the `rel.toLowerCase() === type` comparison, because for `rel = ''`, the `this.refs[ i ].rel &&` would be evaluate to `false`, which in turn would not yield the expected result; e.g. `.has( 'rel', '' )` would always return `false`, even if an entry with an empty `rel` would be present.

Was thinking about cases where this module is used to validate headers, where one might want to check for things like this.

Reply:
Got confused with the impl's of `.rel()` and `.has()` here, but same goes for `.rel( '' )`